core/src/plugins/filed/BareosFdPluginLibcloud.py

```diff
@@ -0,0 +1,386 @@
+                % (self.options["host"], self.options["port"]),
+            )
+            return bRCs["bRC_Error"]
+
+        jobmessage(
+            "M_INFO",
+            "Connected, last backup: %s (ts: %s)" % (self.last_run, self.since),
+        )
+
+        self.api = None
+        try:
+            self.api = BareosLibcloudApi(
+                self.options, self.last_run, self.options["temporary_download_directory"]
+            )
+            debugmessage(100, "BareosLibcloudApi started")
+        except Exception as e:
+            debugmessage(100, "Error: %s" % e)
+            jobmessage("M_FATAL", "Something went wrong with BareosLibcloudApi: %s" % e)
+            return bRCs["bRC_Cancel"]
+
+        return bRCs["bRC_OK"]
+
+    def end_backup_job(self, context):
+        if self.active:
+            self.__shutdown()
+        return bRCs["bRC_OK"]
+
+    def check_file(self, context, fname):
+        # All existing files/objects are passed to bareos
+        # If bareos have not seen one, it does not exists anymore
+        return bRCs["bRC_Error"]
+
+    def __shutdown(self):
+        self.active = False
+        jobmessage(
+            "M_INFO",
+            "BareosFdPluginLibcloud finished with %d files"
+            % (self.number_of_objects_to_backup)
+        )
+
+        debugmessage(100, "Shut down BareosLibcloudApi")
+        self.api.shutdown()
+        debugmessage(100, "BareosLibcloudApi is shut down")
+
+    def start_backup_file(self, context, savepkt):
+        error = False
+        while self.active:
+            if self.api.check_worker_messages() != SUCCESS:
+                self.active = False
+                error = True
+            else:
+                self.current_backup_job = self.api.get_next_job()
+                if self.current_backup_job != None:
+                    break
+                elif self.api.worker_ready():
+                    self.active = False
+                else:
+                    sleep(0.01)
+
+        if not self.active:
+            self.__shutdown()
+            savepkt.fname = "empty"  # dummy value
+            if error:
+                jobmessage("M_FATAL", "Shutdown after worker error")
+                return bRCs["bRC_Cancel"]
+            else:
+                return bRCs["bRC_Skip"]
+
+        filename = FilenameConverter.BucketToBackup(
+            "%s/%s"
+            % (self.current_backup_job["bucket"], self.current_backup_job["name"],)
+        )
+        jobmessage("M_INFO", "Backup file: %s" % (filename,))
+
+        statp = bareosfd.StatPacket()
+        statp.size = self.current_backup_job["size"]
+        statp.mtime = self.current_backup_job["mtime"]
+        statp.atime = 0
+        statp.ctime = 0
+
+        savepkt.statp = statp
+        savepkt.fname = filename
+        savepkt.type = bareos_fd_consts.bFileType["FT_REG"]
+
+        if self.current_backup_job["type"] == JOB_TYPE.DOWNLOADED:
```

Comment:
I think we have a name collision here between "bareos job" and "plugin jobs"? Can this be avoided with different naming?

Reply:
using "task"